pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -189,18 +204,35 @@ func mergeRoutesWeightedClusters(originalRoutes, latestRoutes []*RouteWeightedCl
 	return originalRoutes
 }
 
-// subset returns true if the second array is completely contained in the first array
-func subset(first, second []string) bool {
-	set := make(map[string]bool)
-	for _, value := range first {
-		set[value] = true
+// subset union returns true if either of the slices is a subset of the other and returns a unified slice
```

Comment:
Kindly drop some examples in the function comments. Also what does unified mean - union or intersection here? I'd change the term for that part.

Reply:
I have removed unified and renamed it to union, please refer the unit tests, there are sufficient examples there to explain all scenarios